File: 1260.cpp
#include <stdio.h>
#include <stack>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

vector<int> graph[1001];
int t;
int n,m,s;
int e0,e1;
int visited[1001];

void initialize_visit(){
	for(int i = 1; i <= n; i++){
		visited[i] = 0;
	}
}
void dfs(vector<int> graph[], int start){
	initialize_visit();

	stack<int> s;
	s.push(start);

	while(!s.empty()){
		t = s.top();
		// printf("stack: %d\n", t);
		s.pop();
		if(!visited[t]){
			printf("%d ", t);
			visited[t] = 1;			
		}
		

		for(int i = graph[t].size() - 1; i >= 0; i--){
			if(visited[graph[t][i]]) continue;
			else s.push(graph[t][i]);
		}
	}
	printf("\n");
};

void bfs(vector<int> graph[], int start){
	initialize_visit();
	queue<int> q;
	q.push(start);
	visited[start] = 1;
	while(!q.empty()){
		t = q.front();
		q.pop();
		printf("%d ", t);
		for(int i = 0; i < graph[t].size(); i++){
			// printf("a: %d ", graph[t][i]);
			if(visited[graph[t][i]]) continue;
			else {
				q.push(graph[t][i]);
				visited[graph[t][i]] = 1;
			}
		}
	}
	printf("\n");
}
int main(){
	scanf("%d %d %d",&n,&m,&s);
	for(int i = 0; i < m; i++){
		scanf("%d %d", &e0, &e1);
		if(find(graph[e0].begin(),graph[e0].end(),e1) == graph[e0].end()){
			graph[e0].push_back(e1);
			graph[e1].push_back(e0);
		}
	}

	for(int i= 1; i <= n; i++){
		sort(graph[i].begin(), graph[i].end());

	}

	dfs(graph, s);
	bfs(graph, s);
	return 0;
}

File: 1260.py
import sys
import queue

# import queue
tmp = sys.stdin.readline().split()

class Stack(object):

   def __init__(self):
      self.items = []

   def push(self, item):
      self.items.append(item)

   def pop(self):
       return self.items.pop()

   def peek(self):
       return self.items[-1]

   def isEmpty(self):
       return len(self.items) == 0

graph = {}
for _ in range(int(tmp[1])):
	edge_s = sys.stdin.readline().split()
	edge_s[0] = int(edge_s[0])
	edge_s[1] = int(edge_s[1])
	if edge_s[0] not in graph:
		graph[edge_s[0]] = []
	if edge_s[1] not in graph:
		graph[edge_s[1]] = []

	if edge_s[1] not in graph[edge_s[0]]:
		graph[edge_s[0]].append(edge_s[1])
	if edge_s[0] not in graph[edge_s[1]]:
		graph[edge_s[1]].append(edge_s[0])


def dfs(graph, start):
	s = Stack()
	s.push(start)
	visited = []


	while (not s.isEmpty()):
		_t = s.pop()
		if _t not in visited:
			visited.append(_t)
			inputTostack = list(set(graph[_t]) - set(visited))
			inputTostack.sort(reverse = True)
			# print(inputTostack)
			# inputTostack.reverse()
			for i in range(len(inputTostack)):
				s.push(inputTostack[i])
	return visited

def bfs(graph, start):
	visited = []
	q = queue.Queue()
	q.put(start)
	visited.append(start)
	while (q.qsize()):
		# print(visited)
		_t = q.get(0)
		# if _t in visited:
		# 	continue
		# visited.append(_t)

		inputTostack = list(set(graph[_t]) - set(visited))
		inputTostack.sort(reverse = False)
		# print(inputTostack)
		for i in range(len(inputTostack)):
			if inputTostack[i] not in visited:
				visited.append(inputTostack[i])
				q.put(inputTostack[i])
	return visited

path = dfs(graph,int(tmp[2]))
# print(path)
for p in path:
	print(p, end=' ')
print()
path = bfs(graph,int(tmp[2]))
for p in path:
	print(p, end=' ')
print()